File: 프로그래머스/0/181902. 문자 개수 세기/문자 개수 세기.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(string my_string) {
    // 크기가 52이고 0으로 초기화된 vector 생성
    vector<int> answer(52, 0);
    
    for (char ch : my_string) {
        if (ch >= 'A' && ch <= 'Z') {
            answer[ch - 'A']++;
        } else if (ch >= 'a' && ch <= 'z') {
            answer[ch - 'a' + 26]++;
        }
    }
    
    return answer;
}

File: 프로그래머스/0/181902. 문자 개수 세기/문자 개수 세기.py
def solution(my_string):
    # 길이 52짜리 배열을 0으로 초기화
    answer = [0] * 52
    
    for ch in my_string:
        # 대문자인 경우 ('A'~'Z' -> 인덱스 0~25)
        if 'A' <= ch <= 'Z':
            idx = ord(ch) - ord('A')
            answer[idx] += 1
        # 소문자인 경우 ('a'~'z' -> 인덱스 26~51)
        elif 'a' <= ch <= 'z':
            idx = ord(ch) - ord('a') + 26
            answer[idx] += 1
            
    return answer